proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+
+### Proposed CLI argument and MSBuild property changes
+
+The following changes are proposed for .NET Core 3.0. `build` will be the specific verb discussed in the following text. The behavior is intended to be the same for `publish` unless specifically called out.
+
+#### Framework-dependent architecture-netural applications
+
+Maintain framework-dependent architecture-netural applications as the default for `build`.
+  
+* Produces an application with native dependencies for all supported architectures.
+* Remove the capability to produce EXEs for this type of build.
+* If EXEs are desired for this build type, consider for a later release. It's a hard scenario to get right due to filename conflicts.
+
+ASP.NET and console templates should accept this default.
+
+#### Framework-dependent architecture-specific applications
+
+Expose a new opt-in build type, called architecture-specific framework-dependent.
+
+* Produces an application with native dependencies and an EXE for a single specified architecture.
+* This build type can be selected via
+  * MSBuild: `<ArchitectureSpecific>true</ArchitectureSpecific>`
+  * CLI: `--architecture-specific` or `-a`
+* A RID must be specified (or it is an error). Any of the following options is legal to specify a RID:
+  * MSBuild: `<RuntimeIdentifier>RID</RuntimeIdentifier>`
+  * CLI: `--runtime RID`, or `-r RID`
+  * MSBuild: `<RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>` (means: use RID of SDK; this already works)
+* With this build type, `--self-contained` is `false` by default, even when a RID is specified. `--self-contained` or `--self-contained true` must be specified to produce a self-contained build.
+
+WPF and Windows Forms templates should opt-in to this build type, with the following pattern:
+
+```xml
+<ArchitectureSpecific>true</ArchitectureSpecific>
+<RuntimeIdentifier>win-x86</RuntimeIdentifier>
+```
+
+The `win-x86` RID is proposed as the default to match the 32-bit default for .NET Framework projects. There is a perf benefit and it will be more compatible with existing .NET Framework projects that may have 32-bit only dependencies. It does mean that we need to install the Windows 32-bit .NET Core runtime by default (the 32-bit SDK is not needed). Visual Studio should make it straightforward to switch the default RID to `win-x64`.
+
+Note: It is possible that the RID should be specified a different way that better aligns with the Visual Studio configutation manager. This document doesn't take an opinion on that. it is a detail to work out.
+
+Alternative idea for specifying architecture:
+
+* MSBuild: `<ArchitectureSpecific>RID</ArchitectureSpecific>`
+
+This alternative idea requires one instead of two lines, but also replaces the existing `RuntimeIdentifier` property, which can be viewed as good or bad.
```

Comment:
Bad, imo.

RID is sensible for a few scenarios. Something special for this scenario is not progress.

Reply:
I agree. I changed the text.